### ISSUE TITLE
Error output should specify in which file(s) syntax errors occur

### ISSUE THREAD

**Issue by walterdolce** [hashicorp/terraform#11542]
Hi,

Looks like the file path where syntax errors occur is not exposed in the CLI's UI. Please see below.
Also please note I am aware I forgot to specify the right map key.

This issue is about improving UI/UX (IMO).
```
terraform plan -var-file=vars_development.tfvars
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * At column 1, line 1: output of an HIL expression must be a string, or a single list (argument 3 is TypeMap) in:

${var.environment}-${var.elasticsearch}
```

### Terraform Version
```
Terraform v0.8.5
```

### Affected Resource(s)
N/A

### Terraform Configuration Files
```
# variables.tf
variable "environment" {
    default = "development"
    type = "string"
}

variable "elasticsearch" {
  default = {
    domain_name = "<<elasticsearch_domain_name_here>>"
    instance_count = "<<elasticsearch_instance_count_here>>"
    instance_type = "<<elasticsearch_instance_type_here>>"
    zone_awareness_enabled = "<<zone_awareness_enabled_boolean>>"
  }
  type = "map"
}

# elasticsearch.tf
resource "aws_elasticsearch_domain" "elasticsearch" {
  domain_name = "${var.elasticsearch["domain_name"]}"
  elasticsearch_version = "2.3"
  instance_count = "${var.elasticsearch["instance_count"]}"
  instance_type = "${var.elasticsearch["instance_type"]}"

  tags = {
    Environment = "${var.environment}"
    Domain = "${var.environment}-${var.elasticsearch}"
  }
}

# vars_development.tfvars
elasticsearch = {
  domain_name = "development-elasticsearch"
  instance_count = 1
  instance_type = "t2.micro.elasticsearch"
  zone_awareness_enabled = false
}

```

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
The file where the syntax error occur _should_ be part of the output

### Actual Behavior
The file where the syntax error occur **is not** part of the output

### Steps to Reproduce
```
terraform plan -var-file=vars_development.tfvars
```
### Important Factoids
N/A

### References
N/A

**Comment by apparentlymart** [comment]
Hi @walterdolce! Sorry for the long silence here.

There were quite a few different issues covering this and so we didn't do a good job of keeping up to date on them all, but I'm pleased to report that better error messages are now in master and ready to be included in the forthcoming v0.12.0 release. You can see some examples over in #7189.

Thanks for reporting this!

**Comment by walterdolce** [comment]
That's ok. Thank you!
